proofs/tree.go

```diff
@@ -753,21 +754,20 @@ func newMessageFlattener(message, messageSalts proto.Message, saltsLengthSuffix
 	return &f
 }
 
-func ensurePrefixTrailingDot(prefix string) string {
-	if prefix != "" && !strings.HasSuffix(prefix, ".") {
-		prefix = prefix + "."
-	}
-	return prefix
-}
-
 // FlattenMessage takes a protobuf message struct and flattens it into an array
 // of nodes.
 //
 // The fields are sorted lexicographically by their protobuf field names.
 func FlattenMessage(message, messageSalts proto.Message, saltsLengthSuffix string, hashFn hash.Hash, valueEncoder ValueEncoder, parentPrefix string) (leaves []LeafNode, err error) {
 	f := newMessageFlattener(message, messageSalts, saltsLengthSuffix, hashFn, valueEncoder)
 
-	err = f.generateLeaves(ensurePrefixTrailingDot(parentPrefix), f)
+	var rootProp *Property
+	if parentPrefix != "" {
+		rootProp = &Property{
+			Text: parentPrefix,
```

Comment:
This should be changed to allow supplying both a prefix for the "Text" and bytes.

Reply:
The challenge with the root property is that it only has an `OrdNum` of type `int64`. When creating a tree, we would want to add a prefix that is multiple levels deep. Eg.:

`document.data` => append(unt64(1), uint64(20))

It would be quite cumbersome to create a set of nested `Property` objects just to pass in these details. I think it's better to just ask for `[]uint64` instead.

Reply:
If I change a Property to contain []uint64 instead of a single uint64, then you could specify an arbitrarily long prefix within a single property, and this would let this retain a Property as a prefix. Would you be okay with that? (Side benefit, this would get rid of the ugly *uint64 field in Property, which I think would be nicer as well).

Reply:
I would still use nested properties to concat them, and keep the parent. I think the parent structure is useful.